### PR TITLE
CORE-8991

### PR DESCRIPTION
**Pull request by ianmcorvidae**
Set a new 'unknown' type when a type is not detected; add an 'ipc-info-typer' unit to designate the type was detected by info-typer.

In combination with https://github.com/cyverse-de/ui/pull/201 this should cause no issues. https://github.com/cyverse-de/data-info/pull/13 is the parallel data-info pull request for setting a different unit on user-specified info-types.

### CODE REVIEW

File: src/info_typer/messaging.clj [hunk 1]
```diff
@@ -36,14 +36,14 @@
...
                   ; Double-check an attribute hasn't been added during the time it took for us to detect.
-                  (when-not (or (nil? ctype) (string/blank? ctype) (meta/attribute? cm path (cfg/garnish-type-attribute)))
+                  (when-not (meta/attribute? cm path (cfg/garnish-type-attribute))
                     (log/info "adding type" ctype "to file" id)
-                    (meta/add-metadata cm path (cfg/garnish-type-attribute) ctype "")
-                    (log/debug "done adding type" ctype "to file" id))
-                  (when (or (nil? ctype) (string/blank? ctype))
-                    (log/warn "type was not detected for file" id))))
+                    (meta/add-metadata cm path (cfg/garnish-type-attribute) ctype "ipc-info-typer")
+                    (log/debug "done adding type" ctype "to file" id))))
```
Comment (tedgin) [inline src/info_typer/messaging.clj:46]: I think `ctype` is supposed to be `detected-type`.

**Comment by tedgin** [comment]
Other than the one bad variable name, it looks good to me.

File: src/info_typer/messaging.clj (hunk 1 above)
Comment (ianmcorvidae) [reply src/info_typer/messaging.clj:46]: Not here, no. First it gets `detected-type` using the content-detection algorithm, which might return nil or blank, then the second line of the `let` defines `ctype`, which if the initial detection was nil or blank sets `ctype` to `"unknown"` (and logs that it's done so). Then, the rest of the function operates off of this type.

Basically, if a type wasn't detected and this line used `detected-type`, it'd set a nil or blank value, which we don't want, we want it to set `"unknown"`, which is in `ctype` instead.

**Comment by ianmcorvidae** [comment]
Thanks for the review!

File: src/info_typer/messaging.clj (hunk 1 above)
Comment (tedgin) [reply src/info_typer/messaging.clj:46]: I missed that step.  I'm getting a little rusty with Clojure.

**Pull request merged**